### ISSUE TITLE
assigning an array to a variable of type `Record<string, any>` should be an error

### ISSUE THREAD

**Issue by ivanhofer** [microsoft/TypeScript#50439]
# Bug Report

An array is assignable to a variable of type `Record<string, any>`.
When using any other type instead of `any`, an error gets correctly shown e.g `Record<string, unknown>`.

### 🔎 Search Terms

record, array, any

### 🕗 Version & Regression Information

This issue seems to occur in all `TypeScript` versions.

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/DYUwLgBAHgXBBKIDGB7ATgEwDwGcxoEsA7AcwBoIBDIgTwD4IBeCAbwF8AoKJ1z75gNoBdCAHpREMAAsCOCDikoArsAzzFAdypEIINGnQcOoSDTiJUmXPmLkISogGsiKDUQbN2HGjy8-BQhxAA)

### 💻 Code

```ts
let x: Record<string, any> = {}
x = {}
x = [] // this should show an error

let y: Record<string, unknown> = {}
y = {}
y = [] // correctly shows an error
```

### 🙁 Actual behavior

An array is assignable to a variable of type `Record<string, any>`

### 🙂 Expected behavior

When using `any` for a `Record`, an error shold be shown if an array gets assigned to it.

**Comment by MartinJohns** [comment]
In 2019 this was marked working as intended. See https://github.com/microsoft/TypeScript/issues/33786#issuecomment-538005806.

**Comment by ivanhofer** [comment]
Thanks for pointing me to the comment. Makes sense.

But shouldn't `unknown` then also be valid?

**Comment by RyanCavanaugh** [comment]
The history of this is that people wrote `Record<string, any>` to mean "object" back when `object` didn't exist, so that behavior is preserved for compatibility reason. Then we added `unknown` and `object`, meaning that you can write `Record<string, unknown>` to mean "thing with a string index signature" and `object` to mean, well, `object`.
